cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/pipeline/SystemMetadataWriterStage.java

```diff
@@ -45,7 +48,10 @@ public void process(ApplicationWithPrograms input) throws Exception {
     // add system metadata for apps
     ApplicationId appId = input.getApplicationId();
     ApplicationSpecification appSpec = input.getSpecification();
-    SystemMetadataWriter appSystemMetadataWriter = new AppSystemMetadataWriter(metadataStore, appId, appSpec);
+    // only update creation time if this is a new app
+    Map<String, String> properties = metadataStore.getProperties(MetadataScope.SYSTEM, appId);
+    SystemMetadataWriter appSystemMetadataWriter =
+      new AppSystemMetadataWriter(metadataStore, appId, appSpec, !properties.isEmpty());
```

Comment:
Wouldn't it be cleaner to have the login in AbstractSystemMetadataWriter#write?
Right after calling `getSystemPropertiesToAdd`, remove it from there if its already in `existingProperties`?
It seems like the code changes would be more localized that way.